src/fileseq/filesequence.py

```diff
@@ -150,13 +154,17 @@ def padding(self):
     def setPadding(self, padding):
         """
         Set new padding characters for the sequence.
-        i.e. "#" or "@@@", or an empty string to disable range formatting.
+        i.e. "#" or "@@@" or '%04d', or an empty string to disable range formatting.
 
         :type padding: str
         :rtype: None
         """
         self._pad = padding
-        self._zfill = sum([PAD_MAP[c] for c in self._pad])
+        printf_len = PRINTF_RE.split(self._pad, 1)
+        if len(printf_len) == 3:
+            self._zfill = int(printf_len[1])
+        else:
+            self._zfill = sum([PAD_MAP[c] for c in self._pad])
```

Comment:
I did think about using ```getPaddingNum```, but since I wasn't going to update ```getPaddingChars``` I decided to follow the style and do it at each point. It does make sense use getPaddingNum though, so I'll go ahead update it to use that.

Regarding ```getPaddingChars``` I wasn't 100% sure the best way to approach this since this function is used to convert between numbers and the character representational (which is arbitrarily since you can alter @ or # to represent any number you wanted), while the printf syntax is always going to represent the same value no matter what. Should ```getPaddingChars``` have a flag like ```printf_syntax_padding=True``` or should it be a separate method like ```getPrintfSyntaxFormatting```?